Add doc for auto-installation of missing implementations by Discovery (#301)
php-http/documentation: discovery.rst

Dependency Injection is not an option. Discovery is useful in libraries that want to offer
zero-configuration services relying on the virtual packages.

Auto-installation
-----------------

.. versionadded:: 1.15
    Auto-installation of missing dependencies is available since v1.15.

> Discovery embeds a composer plugin that can auto-install missing implementations.

Reviewer: ```suggestion
Discovery embeds a composer plugin that can auto-install missing implementations when an application does not specify a specific implementation.
```

Author: updated thanks

If a library requires both ``php-http/discovery`` and one of the supported virtual packages (see :doc:`library-developers`),

> the auto-installation logic will select the best matching known implementation when none is already installed.

Reviewer: ```suggestion
but no implementation that provides the virtual package, the auto-installation logic will install the best matching known implementation.
```
